packages/tools/src/tools/annotation/planarFreehandROITool/interpolation/interpolateSegmentPoints.ts

```diff
@@ -0,0 +1,59 @@
+import { Types } from '@cornerstonejs/core';
+import { interpolatePoints } from './algorithms/bspline';
+
+/**
+ * Interpolates a segment of points from iniIndex until endIndex.
+ * It uses the knotSampleSize param to calculate which point from points will be used as knot or not.
+ * Lower knotSampleSize value is more points will be used, i.e the interpolation will be harder.
+ * By default it uses b-spline algorithm.
+ * The result total of points is equal to original points.
+ */
+export default function interpolateSegmentPoints(
+  points: Types.Point2[],
+  iniIndex: number,
+  endIndex: number,
+  knotSampleSize: number
+): Types.Point2[] {
+  const segmentSize = endIndex - iniIndex + 1;
+  if (isNaN(segmentSize) || !segmentSize || !knotSampleSize) {
```

Comment:
You need to check that segmentSize is at least twice the knotSampleSize as you need at least 3 points for interpolation.